src/main/asset_pack.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * TheXTech - A platform game engine ported from old source code for VB6
+ *
+ * Copyright (c) 2009-2011 Andrew Spinks, original VB6 code
+ * Copyright (c) 2020-2024 Vitaly Novichkov <admin@wohlnet.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#ifndef ASSET_PACK_H
+#define ASSET_PACK_H
+
+#include <string>
+#include <vector>
+#include <memory>
+
+#include "std_picture.h"
+
+extern std::string g_AssetPackID;
+
+struct AssetPack_t
+{
+    struct Gfx
+    {
+        StdPicture logo;
+        StdPicture icon;
+        StdPicture background;
+
+        int bg_frames = 1;
+        int bg_frame_ticks = 8;
+    };
+
+    std::string id;
+    std::string version;
+
+    std::string path;
+
+    std::unique_ptr<Gfx> gfx;
+
+    bool show_id = false;
+    bool show_version = false;
+    bool logo_override = false;
+
+    bool operator==(const AssetPack_t& o) const
+    {
+        return this == &o;
```

Comment:
Are you sure this works? Or I miss something... I would add a comparison with every field by field, and return false when any is not equal.

Reply:
This was being used in a single place to delete an asset pack from the list by identity. I moved the logic to the calling site instead for clarity.